app/models/course/assessment/answer/programming.rb

```diff
@@ -56,9 +56,20 @@ def download(dir)
   def assign_params(params)
     acting_as.assign_params(params)
 
+    params[:import_files]&.each do |file|
+      files.build(
+        filename: file.original_filename,
+        content: File.read(file.tempfile.path)
```

Comment:
I'd suggest reading the file on the client side and submit the content instead. So that this special param(`import_files`) is not needed and user can preview the file after uploading as well.